sortinghat/cmd/unify.py

```diff
@@ -155,22 +165,40 @@ def __unify_unique_identities(self, uidentities, matcher,
         self.total = len(uidentities)
         self.matched = 0
 
-        matched = match(uidentities, matcher, fastmode=fast_matching)
+        if self.recovery and RecoveryFile.exists():
+            print("Loading matches from recovery file: %s" % RecoveryFile.location())
+            matched = RecoveryFile.load_matches()
+        else:
```

Comment:
This `if` block means `recovery` is always in use. This code should be under an `if` too.

```
 matched = RecoveryFile.jsonize(matched)
```

Reply:
The `jsonize`method converts the matched identities to a common JSON format, thus it simplifies the logic to deal with matches collected from the recovery file or with the method `match`. However that method will be always used (even if the recovery is not active). 
An option could be to move that method from the `RecoveryFile`class to the `Unify` class. Otherwise, I guess the PR should be significantly changed (see PR #152)